lib/trace-writer.js

```diff
@@ -120,14 +120,10 @@ TraceWriter.prototype.flushBuffer_ = function(projectId) {
   // Privatize and clear the buffer.
   var buffer = this.buffer_;
   this.buffer_ = [];
+  var joined = buffer.join();
+  buffer = '[' + joined.slice(0, joined.length -1) + ']';
 
-  var traces = { traces: buffer };
-  // FIXME(ofrobots): if the buffer is large, this is going to burn
-  // a lot of CPU and block the event loop. Figure out a way to do this some
-  // other way. Since node-core doesn't have a streaming JSON stringifier
-  // we'll need to look at npm modules for this.
-  // Alternatively, we should switch to protobufs
-  this.publish_(projectId, JSON.stringify(traces));
+  this.publish_(projectId, '{"traces":' + buffer + '}');
```

Comment:
Looks like it would be overkill in this context: https://github.com/nodejs/node/blob/master/lib/util.js#L9 it might be cleaner but I don't think it would be more performant